frontend/src/components/Patchnotes.jsx

```diff
@@ -5,224 +5,362 @@ import Navbar from './Navbar';
-                <form
-                    onSubmit={(e) => {
-                        e.preventDefault();
-                        const formData = {
-                            message: e.target.message.value,
-                            patchId: patchnotes.id,
-                            userId: user.id,
-                        };
-                        submitComment(formData);
-                    }}
-                >
-                    <input className='comment-input' type='text' name='message' placeholder='Leave a comment...' required />
-                    <button type='submit' className='comment-button'>Submit</button>
-                </form>
+            {data.abilityUpdates && data.abilityUpdates.length > 0 && (
+              <div>
+                <h4>Ability Updates:</h4>
+                {data.abilityUpdates.map((ability, idx) => (
+                  <div key={idx}>
+                    <h5>{ability.name}</h5>
+                    <ul>
+                      {ability.content.map((detail, i) => (
+                        <li key={i}>{detail}</li>
+                      ))}
+                    </ul>
+                  </div>
+                ))}
+              </div>
+            )}
+            {data.content && data.content.length > 0 && (
+              <div>
+                <h4>Content Updates:</h4>
+                <ul>
+                  {data.content.map((content, idx) => (
+                    <li key={idx}>{content}</li>
+                  ))}
+                </ul>
+              </div>
+            )}
+          </div>
+        );
+      })}
+    </>
+  );
+
+  const renderReplies = (replies) => {
+    return Array.isArray(replies) ? replies.map((reply) => (
+      <div key={reply.id} className='reply'>
+        <p>{reply.user.username}: @{reply.replyTo.username} {reply.message}</p>
+        {renderReplies(reply.replies || [])}
+        <button className='reply-button' onClick={() => openReplyModal(reply.commentId, reply.user.id, reply.id)}>Reply</button>
+      </div>
+    )) : null;
+  };
+
+  return (
+    <div className='patchnotes'>
+      <Navbar />
+      <div className='patches'>
+        {game === 'overwatch' ? (
+          <>
+            {renderCategories('Tank', patchnotes.details.tank)}
+            {renderCategories('Damage', patchnotes.details.damage)}
+            {renderCategories('Support', patchnotes.details.support)}
+            {renderCategories('Map', patchnotes.details.mapUpdates)}
+            {renderCategories('Bug', patchnotes.details.bugFixes)}
+          </>
+        ) : (
+          <>
+            {renderCategories('Agent', patchnotes.details.agentUpdates)}
+            {renderCategories('Map', patchnotes.details.mapUpdates)}
+            {renderCategories('Bug', patchnotes.details.bugFixes)}
+          </>
+        )}
+      </div>
+      <div className='comments'>
+        <h2>Comments</h2>
+        <div>
+          {patchnotes.comments?.map((comment) => (
```

Comment:
again, you may be able to split out the comments functionality into Comments.jsx